engine/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -0,0 +1,89 @@
+package cromwell.services
+
+import akka.actor.Actor
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowMetadata
+import cromwell.engine.{ExecutionEventEntry, WorkflowId}
+import cromwell.engine.backend.JobKey
+import cromwell.services.MetadataServiceActor._
+import cromwell.webservice.WorkflowMetadataResponse
+import org.joda.time.DateTime
+import wdl4s.values.WdlValue
+
+object MetadataServiceActor {
+
+  sealed trait MetadataEvent[V] {
```

Comment:
As I envisioned it, the simplest thing we can do would be a fully-workflow-aware metadata service which takes exactly the right data and stores them in exactly fitting tables (i.e. the already-existing ones). With all the problems which that entails, but it's an easy first-step and it's a service so we can replace it later if necessary